src/services/payments-service.ts

```diff
@@ -111,17 +113,11 @@ export class PaymentsService implements IPaymentsService {
   public async updateInvoice(invoice: Invoice): Promise<void> {
     debug('update invoice %s: %o', invoice.id, invoice)
     try {
+      const fullInvoice = await this.invoiceRepository.findById(invoice.id)
       await this.invoiceRepository.upsert({
-        id: invoice.id,
-        pubkey: invoice.pubkey,
-        bolt11: invoice.bolt11,
-        amountRequested: invoice.amountRequested,
-        description: invoice.description,
-        unit: invoice.unit,
+        ...fullInvoice,
```

Comment:
let's revert this change. `upsert` already updates some invoice properties if one already exists in the database.

Reply:
Does it? I had issues with it iirc

Idk if it's because of not including verify_url here? https://github.com/Cameri/nostream/pull/202/files/6bd859fd43f44f4db0cd3336ca5f7771cd6e5d29#diff-a3487de3d1ff9ff4f5025e4073c92c4f2d51c7b664a818b0fcd38fd5469f5674R127

Reply:
(Testing again)

Reply:
It both updates and inserts. If it's failing with verify_url then it needs to be accounted for.
But I prefer explicitly mapping stuff as opposed to spreading an object because it's unclear what is being mapped.

Reply:
Oops sorry, it's actually because we only have the information to update i.e. status (and id) coming from the getInvoice in the lnurl processor as we don't have access to the whole invoice by id from the server side (since we use a randomUUID)

Reply:
https://github.com/Cameri/nostream/pull/202/files#diff-25cc4285e45b3c547df086050d23de6cca9038dde8a749713e83a28f78f229c7R24-R27

Reply:
@im-adithya should we just have an `updateInvoiceStatus` function?

Reply:
Yes, that would be awesome!

Reply:
Yes, please! We might need a corresponding repository function too?

Reply:
Done! I'm using the upsert by fetching the fullInvoice using Id as before

Reply:
@Cameri just a gentle reminder on this one!

Reply:
You leaked your private keys. Please don't type private keys in the code again and always use environment variables to inject them.

Reply:
Yeah had to remove them :') luckily didn't use them much
> always use environment variables to inject them.

Sure!